### DISCUSSION TITLE
Idea requests: Any additional ideas of features for the module API before release

### DISCUSSION THREAD

**Discussion by ejv2** [DSJAS/DSJAS#28] [Ideas]
As stated in the title, the module API is getting ready to be standardised and released in the next alpha release. I have already written up the basics and shipped the basics, but it doesn't quite feel as feature complete as the theme API yet.

So, what features would people like to see based on the current state. You can see the current state by checking out the ```module-api``` branch, or visiting [here](https://github.com/DSJAS/DSJAS/tree/module-api/public/include/api/modules). The checklist of the basics is in the [original issue board post](https://github.com/DSJAS/DSJAS/issues/27).

Thanks for all the help and support from you guys so far - I really appreciate and it has probably sped up the progress to DSJAS 1.0 by 1000%.

**Comment by SinisterSpatula** [comment]
The only other modules I'm thinking of currently are a bill-pay module (where a user could view their monthly payments to cable, water, electric, etc.   And a zelle or pop money module where you would send payments to people.  And then finally a module for international transfers that could record the transaction info and say something like your transaction will be final by (three days from today) and record the info that was entered into the database for later review.  Anything you think those modules would need (user info, accounts info, etc.)

After that, it would be kind of nice to have a system that updated tranactions, made them age, inserted scheduled new ones (like those of the bill pay module), those are the ideas I've had in my mind for the project, but not sure when I was going to have any time to do them and I'd have a crap ton of learning to do before it became possible.  Aside for that, tutorials, guides, instructions on all the working parts of dsjas for both beginners, intermediate, and advanced are things I would kill for. :D  Great job on all your hard work on this.  It's really raising the bar on what we have available to work with! 🥇

**Comment by SinisterSpatula** [comment]
Oh and an ATM withdrawal module that would insert transactions that looked like an ATM withdrawal (check if the user has the money available, have it come from their checking account, have it add a fee like $2.50 from the bank, and another fee from the ATM owner, have a collection of ATM owners that it would use to generate the details of the transaction, have the amount withdrawn be selectable in multiples of 20.00.  (this one would be on the admin panel, not visible in the normal areas).

**Comment by ejv2** [comment]
Great ideas. From a technical standpoint, all that's *really* holding those ideas back is the formalisation of the module descriptors system. I added some initial documentation on them in the theme and module documentation sections, which should serve developers well. I've also finished support for them in 2/3 of the default themes.

Once that is done, I'm going to go ahead and release this as the next update.

**Comment by ejv2** [comment]
I'm closing this, as we can discuss any final additions to the module API before it is frozen for v1.0
